### PR TITLE
Update Sumologic to use new Create/Update pattern

### PR DESCRIPTION
**Pull request by mccurdyc**
## Proposed Change(s)

* Uses a factory pattern for Creates and Updates.
* Adds test for non-exported API in `_test.go` file.
* Moves exported API test to `_integration_test.go` file.

## Validation

```
$ make test
```

```
git clone git@github.com:mccurdyc/cli.git /tmp/cli && \
(
    cd /tmp/cli && \
    git checkout mccurdyc/logging-test-sumologic && \
    make test && \
    rm -rf /tmp/cli \
)
```

### CODE REVIEW

File: pkg/logging/sumologic/sumologic_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,386 @@
+package sumologic_test
```
Comment (mccurdyc) [inline pkg/logging/sumologic/sumologic_integration_test.go:1]: This is a direct mv sumologic_test.go sumologic_integration_test.go

File: pkg/logging/sumologic/update.go [hunk 2]
```diff
@@ -16,10 +16,12 @@ type UpdateCommand struct {
...
-	Input fastly.GetSumologicInput
-
-	NewName common.OptionalString
+	// required
+	EndpointName string // Can't shaddow common.Base method Name().
+	Version      int
 
+	// optional
+	NewName           common.OptionalString
 	Address           common.OptionalString
```
Comment (mccurdyc) [inline pkg/logging/sumologic/update.go:25]: @phamann TODO: this field is extraneous, but not removing as part of this change.

https://developer.fastly.com/reference/api/logging/sumologic/

Comment (mccurdyc) [reply pkg/logging/sumologic/update.go]: https://github.com/fastly/cli/pull/103/commits/2a0acd2de3e716ffe19c138bab1041bc0ca8252e

Comment (phamann) [reply pkg/logging/sumologic/update.go]: :+1:

Comment (mccurdyc) [reply pkg/logging/sumologic/update.go:25]: I'll pull into a separate PR just for clarity.

Comment (mccurdyc) [reply pkg/logging/sumologic/update.go:25]: https://github.com/fastly/cli/pull/104

**Pull request merged**